source/features/extend-conversation-status-filters.tsx

```diff
@@ -1,80 +1,67 @@
-import React from 'dom-chef';
-import {$, $$} from 'select-dom';
-import CheckIcon from 'octicons-plain-react/Check';
-import elementReady from 'element-ready';
 import * as pageDetect from 'github-url-detection';
 
 import features from '../feature-manager.js';
 import SearchQuery from '../github-helpers/search-query.js';
+import observe from '../helpers/selector-observer.js';
 
-function addMergeLink(): void {
+function getFilterState(): {isMerged: boolean; isUnmerged: boolean} {
+	const locationQuery = SearchQuery.from(location);
+
+	return {
+		isMerged: locationQuery.includes('is:merged'),
+		isUnmerged: locationQuery.includes('is:unmerged'),
+	};
+}
+
+function addMergeLink(lastLink: HTMLAnchorElement): void {
+	// It's shouldn't be added in issues list
 	if (!pageDetect.isPRList()) {
 		return;
 	}
 
-	// The links in `.table-list-header-toggle` are either:
-	//   1 Open | 1 Closed
-	//   1 Total            // Apparently appears with is:merged/is:unmerged
-	for (const lastLink of $$('.table-list-header-toggle.states a:last-child')) {
-		const lastLinkQuery = SearchQuery.from(lastLink);
-
-		if (lastLinkQuery.includes('is:merged')) {
-			// It's a "Total" link for "is:merged"
-			lastLink.lastChild!.textContent = lastLink.lastChild!.textContent.replace('Total', 'Merged');
-			continue;
-		}
-
-		if (lastLinkQuery.includes('is:unmerged')) {
-			// It's a "Total" link for "is:unmerged"
-			lastLink.lastChild!.textContent = lastLink.lastChild!.textContent.replace('Total', 'Unmerged');
-			continue;
-		}
+	const {isMerged, isUnmerged} = getFilterState();
 
+	if (isMerged) {
+		// It's a "Total" link for "is:merged"
+		lastLink.lastChild!.textContent = lastLink.lastChild!.textContent.replace('Total', 'Merged');
+	} else if (isUnmerged) {
+		// It's a "Total" link for "is:unmerged"
+		lastLink.lastChild!.textContent = lastLink.lastChild!.textContent.replace('Total', 'Unmerged');
+	} else {
 		// In this case, `lastLink` is expected to be a "Closed" link
 		const mergeLink = lastLink.cloneNode(true);
 		mergeLink.textContent = 'Merged';
-		mergeLink.classList.toggle('selected', SearchQuery.from(location).includes('is:merged'));
-		mergeLink.href = SearchQuery.from(mergeLink).replace('is:closed', 'is:merged').href;
+		mergeLink.classList.toggle('selected', isMerged);
+		// If link is selected, the filters are already removed
+		mergeLink.href = lastLink.classList.contains('selected') ? SearchQuery.from(mergeLink).append('is:merged').href : SearchQuery.from(mergeLink).replace('is:closed', 'is:merged').href;
 		lastLink.after(' ', mergeLink);
 	}
 }
 
-function togglableFilters(): void {
-	for (const link of $$('.table-list-header-toggle.states a')) {
-		$('.octicon', link)?.remove();
```

Comment:
The reason of removing this line is it casues a bad experience when it's loading. And I think the original icon is not bad we don't need to remove.

Reply:
The reason it's there is because we're changing the way the widget works: it's being turned into a radio/checkbox